test/jdk/java/awt/Graphics/TextAAHintsTest.java

```diff
@@ -221,26 +196,40 @@ public static void createTestUI() {
         frame.setVisible(true);
     }
 
+    public static void readFailedReason() {
+        // Show dialog to read why the testcase was failed and append the
+        // testcase failure reason to the output
+        final Dialog dialog = new Dialog(frame, "TestCase" +
+                " failure reason", true);
+        TextArea textArea = new TextArea("", 5, 60, TextArea.SCROLLBARS_BOTH);
+        dialog.add(textArea, BorderLayout.CENTER);
+
+        Button okButton = new Button("OK");
+        okButton.addActionListener(e1 -> {
+            failureReason = textArea.getText();
+            dialog.dispose();
+        });
+        Panel ctlPanel = new Panel();
+        ctlPanel.add(okButton);
+        dialog.add(ctlPanel, BorderLayout.SOUTH);
+        dialog.setLocationRelativeTo(null);
+        dialog.pack();
+        dialog.setVisible(true);
+    }
+
     public static void main(String[] args) throws InterruptedException, InvocationTargetException {
-        createTestUI();
-        mainThread = Thread.currentThread();
-        try {
-            mainThread.sleep(testTimeOut);
-        } catch (InterruptedException ex) {
-            if (!testPassed) {
-                throw new RuntimeException("Test failed : Reason : " + failureReason);
-            }
-        } finally {
-            SwingUtilities.invokeAndWait(()->{
-                if ( frame != null) {
+        java.awt.EventQueue.invokeAndWait(TextAAHintsTest::createTestUI);
```

Comment:
I wonder why `EventQueue` uses fully qualified class name instead of being imported as other classes. That's fine, I'm just wondering.